pinotdb/db.py

```diff
@@ -388,14 +388,43 @@ def finalize_query_payload(
             return {"sql": query}
 
     def normalize_query_response(self, input_query, query_response):
+        if query_response.status_code == 400:
+            raise exceptions.ServerError(
+                f"Server returned a 400 Bad Request response."
+            )
+        elif query_response.status_code == 500:
+            raise exceptions.ServerError(
+                f"Server returned a 500 Internal Server Error response."
+            )
+        elif query_response.status_code == 503:
+            raise exceptions.ServiceUnavailableError(
+                f"Server returned a 503 Service Unavailable response."
+            )
+
         try:
             payload = query_response.json()
         except Exception as e:
-            raise exceptions.DatabaseError(
-                f"Error when querying {input_query} from {self.url}, "
+            raise exceptions.MalformedQueryResponseError(
+                f"Malformed response when querying {input_query} from {self.url}, "
                 f"raw response is:\n{query_response.text}"
             ) from e
 
+
+        for exception in payload.get("exceptions", []):
+            if "message" in exception and exception["message"]:
+                if "TableDoesNotExistError" in exception["message"]:
+                    raise exceptions.NotFoundError(
+                        f"Table of query {input_query} not found!"
+                    )
+                elif "UnknownColumnError" in exception["message"]:
+                    raise exceptions.NotFoundError(
+                        f"Column of query {input_query} not found!"
+                    )
+                else:
+                    raise exceptions.InternalError(
+                        exception["message"]
+                    )
+
```

Comment:
A minor readable improvement

```suggestion

        _exception_msgs = [exception["message"] for exception in payload.get("exceptions", []) if "message" in exception]
        for msg in _exception_msgs:
            if "TableDoesNotExistError" in msg:
                raise exceptions.NotFoundError(
                    f"Table of query {input_query} not found!"
                )
            elif "UnknownColumnError" in msg:
                raise exceptions.NotFoundError(
                    f"Column of query {input_query} not found!"
                )
            else:
                raise exceptions.InternalError(msg)

```